src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionsHandler.cs

```diff
@@ -61,95 +58,11 @@ internal class CodeActionsHandler : AbstractRequestHandler<LSP.CodeActionParams,
                 return Array.Empty<VSCodeAction>();
             }
 
-            var (codeFixCollections, codeRefactorings) = await CodeActionHelpers.GetCodeFixesAndRefactoringsAsync(
-                document, _codeFixService, _codeRefactoringService,
+            var codeActions = await CodeActionHelpers.GetVSCodeActionsAsync(
+                request, document, _codeFixService, _codeRefactoringService, _threadingContext,
                 request.Range, cancellationToken).ConfigureAwait(false);
 
-            var codeFixes = codeFixCollections.SelectMany(c => c.Fixes);
-            using var _ = ArrayBuilder<VSCodeAction>.GetInstance(out var results);
-
-            // Go through code fixes and code refactorings separately so that we can properly set the CodeActionKind.
-            foreach (var codeFix in codeFixes)
```

Comment:
All of this logic is now either handled in CodeActionHelpers or UnifiedSuggestedActionSource.